### PR TITLE
Rnog eventbrowser

### PR DESCRIPTION
**Pull request by shallmann**
Contains modules for rno-g data i/o and improvements to the eventbrowser (w/ possibility to read rno-g files)

We would like to merge this PR since other PRs already rely on the uproot reading capabilities. We want to cleanup of the rnogDataReader and the readRNOGData scripts (similar functionality as in eventReader and NuRadioRecoio) after this PR.

### CODE REVIEW

File: NuRadioReco/detector/detector_browser/index.py [hunk 1]
```diff
@@ -36,4 +36,4 @@
     if int(dash.__version__.split('.')[0]) <= 1:
         if int(dash.__version__.split('.')[1]) < 0:
             print('WARNING: Dash version 0.39.0 or newer is required, you are running version {}. Please update.'.format(dash.__version__))
-    app.run_server(debug=False, port=8080)
+    app.run_server(debug=True, port=8082)
```
Comment (anelles) [inline NuRadioReco/detector/detector_browser/index.py:39]: Are you sure that you want the default to be in debug mode?

File: NuRadioReco/eventbrowser/apps/simulation.py [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-import dash_html_components as html
+from dash import html
 import NuRadioReco.eventbrowser.apps.simulation_plots.sim_electric_field_trace
 import NuRadioReco.eventbrowser.apps.simulation_plots.sim_electric_field_spectrum
 import NuRadioReco.eventbrowser.apps.simulation_plots.sim_event_overview
 import logging
 
 logger = logging.getLogger('traces')
-
+logger.setLevel(logging.DEBUG)
```
Comment (anelles) [inline NuRadioReco/eventbrowser/apps/simulation.py:9]: Same question as above. Should this really be the default?

File: NuRadioReco/eventbrowser/apps/simulation_plots/sim_event_overview.py [hunk 3]
```diff
@@ -35,8 +38,12 @@ def update_sim_event_3d(i_event, filename, station_id, juser_id):
     ariio = provider.get_arianna_io(user_id, filename)
```
Comment (anelles) [inline NuRadioReco/eventbrowser/apps/simulation_plots/sim_event_overview.py:38]: Is this ARIANNA related or is this a very unfortunate remainder? 
Can also be cleaned up later, if this is harder to fix.

File: NuRadioReco/eventbrowser/apps/trace_plots/multi_channel_plot.py [hunk 4]
```diff
@@ -69,6 +78,36 @@
     dcc.Graph(id='time-traces')
 ]
 
+# @app.callback(
```
Comment (anelles) [inline NuRadioReco/eventbrowser/apps/trace_plots/multi_channel_plot.py:81]: Should this be removed?

File: NuRadioReco/eventbrowser/dataprovider_root.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    def __new__(cls):
+        if DataProviderRoot.__instance is None:
+            DataProviderRoot.__instance = object.__new__(cls)
+        return DataProviderRoot.__instance
+
+    def __init__(self):
+        self.__user_instances = {}
+
+    def get_arianna_io(self, user_id, filename):
```
Comment (anelles) [inline NuRadioReco/eventbrowser/dataprovider_root.py:15]: Somehow this seems like this is mixing RNO-G and ARIANNA now ... not sure I completely understand.

File: NuRadioReco/eventbrowser/index.py [hunk 6]
```diff
@@ -170,46 +180,63 @@ def get_page_content(selection):
...
         if context.triggered[0]['prop_id'] == 'btn-previous-event.n_clicks_timestamp':
             if i_event == 0:
-                return 0
+                event_i = 0
             else:
-                return i_event - 1
+                event_i = i_event - 1
         if context.triggered[0]['prop_id'] == 'btn-next-event.n_clicks_timestamp':
-            user_id = json.loads(juser_id)
+            # user_id = json.loads(juser_id)
```
Comment (anelles) [inline NuRadioReco/eventbrowser/index.py:232]: Here (and further below) there are a bunch on uncommented lines. Are they here for a reason (work in progress) or should they rather go?

**Review by anelles — DISMISSED** [review]
Very nice work. Thanks! I left a couple of cosmetic comments of what I noticed. I did not test this in detail, as the PR said that this is work in progress anyway and needs to be merged fast. 
Please take a look at the ARI vs RNO-G comments. If this ends up in a released, I think we want this cleaned up rather sooner than later to avoid confusion.

File: NuRadioReco/eventbrowser/dataprovider_root.py (hunk 5 above)
Comment (sjoerd-bouma) [reply NuRadioReco/eventbrowser/dataprovider_root.py:15]: We inherited this from the previous eventbrowser - the file access function was called get_arianna_io. It should be relatively straightforward to rename it to something more neutral everywhere, e.g. get_io_handler

Comment (anelles) [reply NuRadioReco/eventbrowser/dataprovider_root.py]: Thanks for fixing this. This has actually been around for way too long. The files were called .ari, before we decieded for something more general as .nur

**Review by anelles — DISMISSED** [review]
I did not test it on files .. but code looks fine from my side now.

**Review by shallmann — COMMENTED** [review]
minor cleanup. Thanks @sjoerd-bouma for taking over remaining modifications!

**Comment by shallmann** [comment]
Sorry @anelles, you have to approve again, I dismissed your approval because of some minor final cosmetic changes (`import XX as XX` -> `import XX`)

File: NuRadioReco/eventbrowser/apps/overview_plots/channel_properties.py [hunk 7]
```diff
@@ -1,16 +1,25 @@
...
+from dash import html
+from dash import dcc
 from dash.dependencies import Input, Output, State
 from NuRadioReco.utilities import units
 from NuRadioReco.framework.parameters import channelParameters as chp
 import NuRadioReco.eventbrowser.apps.common
 import NuRadioReco.eventbrowser.dataprovider
+import logging
 
+logger = logging.getLogger('overview')
```
Comment (lpyras) [inline NuRadioReco/eventbrowser/apps/overview_plots/channel_properties.py:12]: Is it correct that no logger.setLevel(logging.WARNING) or something similar is set? Maybe you can add a line here if it is not inherited.

File: NuRadioReco/eventbrowser/apps/overview_plots/event_overview.py [hunk 8]
```diff
@@ -2,15 +2,18 @@
...
 from NuRadioReco.utilities import units
 from NuRadioReco.framework.parameters import stationParameters as stnp
 from NuRadioReco.framework.parameters import eventParameters as evp
 from NuRadioReco.framework.parameters import showerParameters as shp
 import radiotools.helper as hp
 import NuRadioReco.eventbrowser.dataprovider
 import numpy as np
+import logging
+
+logger = logging.getLogger('overview')
```
Comment (lpyras) [inline NuRadioReco/eventbrowser/apps/overview_plots/event_overview.py:16]: maybe set a logging level here.

File: NuRadioReco/eventbrowser/apps/overview_plots/trigger_properties.py [hunk 9]
```diff
@@ -32,10 +33,14 @@ def trigger_overview_properties(filename, evt_counter, station_id, juser_id):
...
         for setting_name in trigger.get_trigger_settings():
+            display_value = '{}'
+            setting_value = trigger.get_trigger_settings()[setting_name]
+            if type(setting_value) in [float, np.float32, np.float64, np.float128]:
+                display_value = '{:.5g}'
             props.append(
                 html.Div([
                     html.Div('{}'.format(setting_name), className='custom-table-td'),
-                    html.Div('{}'.format(trigger.get_trigger_settings()[setting_name]),
+                    html.Div(display_value.format(setting_value),
```
Comment (lpyras) [inline NuRadioReco/eventbrowser/apps/overview_plots/trigger_properties.py:43]: I am bad in html. Is that the correct format?

File: NuRadioReco/eventbrowser/apps/simulation.py [hunk 10]
```diff
@@ -1,11 +1,13 @@
...
-import dash_html_components as html
+from dash import html
 import NuRadioReco.eventbrowser.apps.simulation_plots.sim_electric_field_trace
 import NuRadioReco.eventbrowser.apps.simulation_plots.sim_electric_field_spectrum
 import NuRadioReco.eventbrowser.apps.simulation_plots.sim_event_overview
 import logging
 
 logger = logging.getLogger('traces')
+parent_logger = logging.getLogger('NuRadioReco')
+logger.setLevel(parent_logger.level)
```
Comment (lpyras) [inline NuRadioReco/eventbrowser/apps/simulation.py:10]: nice :)

File: NuRadioReco/eventbrowser/apps/simulation_plots/sim_event_overview.py [hunk 11]
```diff
@@ -3,13 +3,16 @@
...
-import dash_html_components as html
-import dash_core_components as dcc
+from dash import html
+from dash import dcc
 from dash.dependencies import Input, Output, State
 from NuRadioReco.eventbrowser.app import app
 import NuRadioReco.eventbrowser.dataprovider
+import logging
 
+logger = logging.getLogger('traces')
```
Comment (lpyras) [inline NuRadioReco/eventbrowser/apps/simulation_plots/sim_event_overview.py:15]: set level?

**Review by lpyras — COMMENTED** [review]
Thanks for doing this!

**Review by anelles — APPROVED** [review]
Also small changes don't go unnoticed ;)

File: NuRadioReco/eventbrowser/apps/overview_plots/channel_properties.py (hunk 7 above)
Comment (sjoerd-bouma) [reply NuRadioReco/eventbrowser/apps/overview_plots/channel_properties.py:12]: Loggers are created with level [logging.WARNING by default](https://docs.python.org/3/howto/logging.html#when-to-use-logging). I guess I can make all of them inherit from the NuRadioReco logger in index.py (which is currently set to logging.INFO), though.

Comment (lpyras) [reply NuRadioReco/eventbrowser/apps/overview_plots/channel_properties.py:12]: Thank you for the clarification :)

File: NuRadioReco/eventbrowser/apps/overview_plots/trigger_properties.py (hunk 9 above)
Comment (shallmann) [reply NuRadioReco/eventbrowser/apps/overview_plots/trigger_properties.py:43]: yes, looks good

**Comment by shallmann** [comment]
good you spotted that hudl-mistake @sjoerd-bouma

**Review by lpyras — APPROVED** [review]
Thanks!

**Comment by shallmann** [comment]
We will keep the branch for now, in case further development for the monitoring tool is needed.

**Pull request merged**